include/swift/AST/DiagnosticsSema.def

```diff
@@ -2454,8 +2454,9 @@ NOTE(overridden_near_match_here,none,
      "potential overridden %0 %1 here",
       (DescriptiveDeclKind, DeclName))
 ERROR(override_decl_extension,none,
-      "overriding %select{|non-@objc }0declarations "
-      "%select{in extensions|from extensions}0 is not supported", (bool, bool))
+      "%select{|non-@objc}0 method %2 %select{"
```

Comment:
We have `DescriptiveDeclKind` that can print out a more accurate descriptor than `method` when e.g. a constructor is the subject of an override.

Reply:
Also, I don't believe the @objc-ness of a method is a salient detail here.

Reply:
If the method is not @objc, there is an additional diagnosis that suggests the user make the method @objc so that it can be overridden in extensions. I think adding some detail to this diagnosis as well might help. What do you think?